plugins/repository-hdfs/hadoop-common/build.gradle

```diff
@@ -0,0 +1,10 @@
+apply plugin: 'elasticsearch.java'
+apply plugin: 'com.github.johnrengelman.shadow'
+
+dependencies {
+  implementation "org.apache.hadoop:hadoop-common:${project.parent.versions.hadoop2}"
+}
+
+tasks.named('shadowJar').configure {
+  exclude 'org/apache/hadoop/util/ShutdownHookManager$*.class'
```

Comment:
Is the dollar correct? We don't want the actual ShutdownHookManager either, not just subclasses

Reply:
This filtering is done on the "final" jar, meaning that would rip out our override class. The ShadowJar copy action looks to avoid duplicate archive entries though so I think it's safe to presume this behavior. If for some reason this would change we'd observe things explode due to those missing inner classes anyhow.